contrib/fdb-c-cmake/CMakeLists.txt

```diff
@@ -0,0 +1,143 @@
+set(FDB_VERSION 7.1.31)
+set(FDB_API_VERSION 710)
+
+if(OS_LINUX)
+    set(FOUNDATIONDB_SOURCE_PATCHS
+        inject-fdb-c-extensions.patch
+        only-build-client.patch
+        fix-harmful-symbol.patch
+        fix-cmake-flow-target.patch
+        backport-doctest-2.4.8.patch
+        fix-missing-ftw.patch
+        disable-s3.patch
+        bundle-llvm-runtimes.patch
+        fix-jemalloc-dlopen.patch
+    )
+    list(TRANSFORM FOUNDATIONDB_SOURCE_PATCHS PREPEND ${CMAKE_CURRENT_SOURCE_DIR}/)
+    set(FOUNDATIONDB_C_EXTENSION_SOURCE fdb_c.actor.cpp)
+
+    set(FOUNDATIONDB_CMAKE_ARGS
+        -DUSE_CCACHE=ON
+        -DCMAKE_BUILD_TYPE=${CMAKE_BUILD_TYPE}
+        -DCMAKE_INSTALL_PREFIX=<INSTALL_DIR>
+        -DCMAKE_EXPORT_COMPILE_COMMANDS=ON
+        -DCMAKE_TOOLCHAIN_FILE=${CMAKE_TOOLCHAIN_FILE}
+        -DDISABLE_TLS=ON # FIXME
+    )
+    if(SANITIZE STREQUAL "memory")
+        list(APPEND FOUNDATIONDB_CMAKE_ARGS
+            -DUSE_MSAN=ON
+            # FIXME: msan for dynamic tls
+            # MSAN reports false-positive with dynamic TLS. So we have to use
+            # static tls for dlopen libraries.
+            # See alse:
+            #   * https://github.com/llvm/llvm-project/issues/52872
+            #
+            # However, fdb's tls are too big. You have to increase the amount
+            # of surplus static TLS to make dlopen work, which requires glibc >= 2.33:
+            #   export GLIBC_TUNABLES=glibc.rtld.optional_static_tls=10240
+            # Otherwise, dlopen will failed with `cannot allocate memory in static TLS block`
+            # See also:
+            #   * https://maskray.me/blog/2021-02-14-all-about-thread-local-storage
+            #   * https://www.gnu.org/software/libc/manual/html_node/Dynamic-Linking-Tunables.html#index-glibc_002ertld_002enns
+            
+            # Other CXX flags:
+            # * `-DMEMORT_SANITIZER` should be defined in fdb's CMakeLists. It's an upstream bug.
+            "-DCMAKE_CXX_FLAGS=-ftls-model=initial-exec -DMEMORY_SANITIZER"
+        )
+    endif()
+    
+    if(SANITIZE STREQUAL "thread")
+        list(APPEND FOUNDATIONDB_CMAKE_ARGS -DUSE_TSAN=ON)
+    endif()
+
+    include(ExternalProject)
+    ExternalProject_Add(
+        foundationdb_ep
+        GIT_REPOSITORY https://github.com/apple/foundationdb.git
```

Comment:
We don't allow cloning repositories at the build stage.

Reply:
Thank you for your review. Will it cause the compilation to fail? According to `run.log`, the repository seems to be successfully cloned. If not, we are going to continue running the workflow, and then solve the compilation problem (static linking and this problem) in parallel. In addition, since fdb depends on [mono](https://www.mono-project.com/) when compiling (which should be the reason of the failure of fast test compilation), we modified the docker file of fast test in the latest commit.